Community_Lighting.js

```diff
@@ -1781,9 +1783,18 @@ Imported[Community.Lighting.name] = true;
       }
 
 
-      grad.addColorStop(brightness, color1);
-
-
+      if (useSmootherLights) {
+        for (let i = brightness; i < 1; i += 0.1) {
+          let data = hexToRgb(color1);
+          let newRed = data.r - (i * 100 * 2.55);
+          let newGreen = data.g - (i * 100 * 2.55);
+          let newBlue = data.b - (i * 100 * 2.55);
+          grad.addColorStop(i, rgba(newRed, newGreen, newBlue, 1 - i));
+        }
+      }
+      else {
+        grad.addColorStop(brightness, color1);
+      }
```

Comment:
I feel like this is a great start to adding a smooth gradient. I'm not that experienced in Javascript but is there not a math library to create a linear interpolation already?

Maybe this will help. 
https://blog.idrisolubisi.com/how-to-create-beautiful-gradients-with-javascript

Reply:
The `grad = context.createRadialGradient(x1, y1, r1, x1, y1, r2);` should already create a gradient does this not allow for alphas? Is it possible to create an override method for the `createRadialGradient` that will change alpha for us so instead of having `if (useSmootherLights)` used everywhere we use the new override?

Reply:
While it does allow for alphas; with just one color stop I would unfortunately end up with more of a cone shape near the edges. I would prefer smoother edges. And it would be extremely bad to use filters I noticed as I remembered that it would have to have updates from the browser. Manually setting the rgba values for the color stops is the best option to approximate a Gaussian blur without the awful drop in performance. And while I am not against overriding the createRadialGradient method; I'd rather not if I don't necessarily have to.

Reply:
Okay sounds good. It is still  good idea to separate it into an additional method to call out to. 
My syntax may be incorrect. I'm still learning javascript. 
`
// you could just use this and call the method
grad.gradientFade(useSmootherLights);

// it would be nice to just have an extension of the gradient class but if not just a public method in this file returning a gradient
func gradientFade(useSmootherLights: bool) {
    if (useSmootherLights) {
        for (let i = brightness; i < 1; i += 0.1) {
          let data = hexToRgb(color1);
          let newRed = data.r - (i * 100 * 2.55);
          let newGreen = data.g - (i * 100 * 2.55);
          let newBlue = data.b - (i * 100 * 2.55);
          self.addColorStop(i, rgba(newRed, newGreen, newBlue, 1 - i));
        }
      }
      else {
        self.addColorStop(brightness, color1);
      }
 }
`

Reply:
I saw you fixed the bug I introduced with lxl (with a lowercase 'L') who should have been lx1 (with a '1').
Good job.